### PR TITLE
Fix types in JoinStepForm.vue

### PR DESCRIPTION
**Pull request by vdestraitt**
Everything is in the title. The linter was not happy....

### CODE REVIEW

File: src/components/stepforms/JoinStepForm.vue [hunk 1]
```diff
@@ -52,7 +52,12 @@ import { VQBModule } from '@/store';
 import BaseStepForm from './StepForm.vue';
 import Multiselect from './widgets/Multiselect.vue';
 
-const joinTypes = JoinStepFormSchema.properties.type.enum;
+const joinTypes = JoinStepFormSchema.properties.type.enum as JoinStep['type'][];
+
+interface DropdownOption {
+  label: string;
+  trackBy: string;
+}
```
Comment (davinov) [inline src/components/stepforms/JoinStepForm.vue:60]: :+1:

File: src/components/stepforms/JoinStepForm.vue [hunk 2]
```diff
@@ -105,10 +110,10 @@ export default class JoinStepForm extends BaseStepForm<JoinStep> {
     return this.availableDatasetNames.map(name => {
       const option = { label: name, trackBy: name };
       if (this.referencingPipelines.includes(name)) {
-        option['$isDisabled'] = true;
-        option[
-          'tooltip'
-        ] = `Circular reference: you cannot combine ${name} because it references the current dataset.`;
+        Object.assign(option, {
+          $isDisabled: true,
+          tooltip: `Circular reference: you cannot combine ${name} because it references the current dataset.`,
```
Comment (davinov) [inline src/components/stepforms/JoinStepForm.vue:115]: Does that means we should add `$isDisabled` and `tooltip` as optional members of the `DropdownOption` interface?

File: src/components/stepforms/JoinStepForm.vue [hunk 3]
```diff
@@ -77,14 +82,14 @@ export default class JoinStepForm extends BaseStepForm<JoinStep> {
   joinColumns = JoinColumns;
   joinTypes: JoinStep['type'][] = joinTypes;
 
-  get rightPipeline(): object {
+  get rightPipeline(): DropdownOption {
     return {
-      label: this.editedStep.right_pipeline,
-      trackBy: this.editedStep.right_pipeline,
+      label: this.editedStep.right_pipeline as string,
+      trackBy: this.editedStep.right_pipeline as string,
```
Comment (davinov) [inline src/components/stepforms/JoinStepForm.vue:88]: Yes, this is quite unconfortable as  we do not have different types for steps with pipeline references and steps with compete pipelines. I think we should create an issue in that.

**Review by davinov — COMMENTED** [review]
Thanks for clarifying types in this file :+1:

File: src/components/stepforms/JoinStepForm.vue (hunk 2 above)
Comment (vdestraitt) [reply src/components/stepforms/JoinStepForm.vue:115]: True!

**Review by davinov — APPROVED** [review]

**Pull request merged**